### DISCUSSION TITLE
[Question] How to use resx translations in combination with stronly typed resource

### DISCUSSION THREAD

**Discussion by XanatosX** [AvaloniaUI/Avalonia#15458] [General]
Hey there,

I already asked this on [Reddit ](https://www.reddit.com/r/AvaloniaUI/comments/1c73se9/question_how_to_use_the_resx_translation_in/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) some days ago but still got no anwser.

I did switch from Visual Studio to VSCode reworking the approach on how the `Resources.Designes.cs` fiels are getting generated based on [this article](https://www.paraesthesia.com/archive/2022/09/30/strongly-typed-resources-with-net-core/). The process does work as expected and I'm able to load the resx information directly from a code side approach. 
But all my bindings inside the views of Avalonia did break. The static link is not getting resolved anymore leading to the following issue.

```
E:\Reporitories\ModularToolManager\src\ModularToolManager\App.axaml(26,93,26,93): Avalonia error AVLN:0004: Unable to resolve Application_Name as static field, property, constant or enum value Line
 26, position 93. [E:\Reporitories\ModularToolManager\src\ModularToolManager\ModularToolManager.csproj]
```

For additional information how the [`csproj`](https://github.com/XanatosX/ModularToolManager/blob/feature/123_create-designer-files-without-visual-studio/src/ModularToolManager/ModularToolManager.csproj) file or the [`app.axaml`](https://github.com/XanatosX/ModularToolManager/blob/feature/123_create-designer-files-without-visual-studio/src/ModularToolManager/App.axaml) is configured please take a look at my [current branch](https://github.com/XanatosX/ModularToolManager/tree/feature/123_create-designer-files-without-visual-studio) for the changes.

As a workaround I could rework my current approach of translation to use bindings with baking fields instead of the `x:static` binding, but I would rather continue working with the current implementation for translation, which is also how it is listed on the Avalonia Wiki.

Thank you in advance

**Comment by timunie** [comment]
Make sure your resx generator is switched to public.

**Comment by XanatosX** [comment]
I did change my ItemGroup from

```
  <ItemGroup>
      <EmbeddedResource Update="Properties\Resources.resx">
      <Generator>MSBuild:Compile</Generator>
      <LastGenOutput>Resources.Designer.cs</LastGenOutput>
      <StronglyTypedFileName>$(IntermediateOutputPath)\Resources.Designer.cs</StronglyTypedFileName>
      <StronglyTypedLanguage>CSharp</StronglyTypedLanguage>
      <StronglyTypedNamespace>ModularToolManager.Properties</StronglyTypedNamespace>
      <StronglyTypedClassName>Resources</StronglyTypedClassName>
    </EmbeddedResource>
  </ItemGroup>
```

to

```
  <ItemGroup>
      <EmbeddedResource Update="Properties\Resources.resx">
      <Generator>MSBuild:Compile</Generator>
      <PublicClass>true</PublicClass>
      <LastGenOutput>Resources.Designer.cs</LastGenOutput>
      <StronglyTypedFileName>$(IntermediateOutputPath)\Resources.Designer.cs</StronglyTypedFileName>
      <StronglyTypedLanguage>CSharp</StronglyTypedLanguage>
      <StronglyTypedNamespace>ModularToolManager.Properties</StronglyTypedNamespace>
      <StronglyTypedClassName>Resources</StronglyTypedClassName>
    </EmbeddedResource>
  </ItemGroup>
```

and it worked immediately. I just missed the `<PublicClass>true</PublicClass>` part. Thank you very much.

**Reply by timunie** [reply]
please also update or link the reddit question, just in case anyone else find it.

**Reply by XanatosX** [reply]
Thank you I did link this discussion and added the solution as a small snippet.
